src/com/amazon/kinesis/streaming/agent/tailing/AbstractParser.java

```diff
@@ -465,6 +469,13 @@ private void getNewCurrentBuffer() {
 
     private R buildRecord(int offset, int length) {
         ByteBuffer data = ByteBuffers.getPartialView(currentBuffer, offset, length);
+
+        final Matcher fileFooterMatcher = fileFooterPattern.matcher(data.asCharBuffer());
```

Comment:
this might be a performance issue though if we want to do this regex match for each record

Reply:
I could do a null check on fileFooterPattern to see if it has been set. That way the test would only happen if configured. 
This feature would mainly be useful for small or short lived files, eg in the case of files getting ftp-ed to a server periodically. The performance hit here would be more than made up for by the performance gain of not tracking a lot of small files. @chaochenq What do you think?

Reply:
sounds good to me. Yea if the null check of fileFooterPattern is set, the performance won't be affected for existing customers. I like this idea and thanks for doing this!